server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -505,23 +504,25 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
      * @return int representing precision
      */
     public static int parsePrecision(XContentParser parser) throws IOException, ElasticsearchParseException {
-        XContentParser.Token token = parser.currentToken();
-        if (token.equals(XContentParser.Token.VALUE_NUMBER)) {
-            return XContentMapValues.nodeIntegerValue(parser.intValue());
-        } else {
-            String precision = parser.text();
+        return parser.currentToken() == Token.VALUE_NUMBER ? parser.intValue() : parsePrecisionString(parser.text());
+    }
+
+    /**
+     * Attempt to parse geohash precision string into an integer value
+     */
+    public static int parsePrecisionString(String precision) {
+        try {
+            // we want to treat simple integer strings as precision levels, not distances
+            return checkPrecisionRange(Integer.parseInt(precision));
+            // Do not catch IllegalArgumentException here
```

Comment:
Can you explain why in the code comment please?

Reply:
Can you add this to the code comment please so its clear for people coming across this in the future?